lib/Controller/Id4meController.php

```diff
@@ -173,11 +179,16 @@ public function code($state = '', $code = '', $scope = '') {
 		$params = $this->request->getParams();
 
 		if ($this->session->get(self::STATE) !== $state) {
+			$responseData = [
+				'error' => 'invalid_state',
+				'error_description' => 'received state does not match the expected value',
+			];
+			if ($this->config->getSystemValueBool('debug', false)) {
+				$responseData['got'] = $state;
+				$responseData['expected'] = $this->session->get(self::STATE);
+			}
 			// TODO show page with forbidden
```

Comment:
it's returning a forbidden, so we can remove this? It shows an error in the frontend?

Reply:
I guess the comment means "returning a proper forbidden page with a nice template".
Now it only shows the raw returned json.

Reply:
Maybe we can move to an error template page directly as seeing json in the browser is far from nice ;)

Reply:
:+1: Done